src/connection.rs

```diff
@@ -705,38 +705,31 @@ impl Connection {
             return;
         }
 
-        self.update_client_quality_score();
+        self.refresh_client_conn_quality_score();
     }
 
-    /// Updates [`ClientConnectionQualityScore`] of this [`Connection`].
-    fn update_client_quality_score(&self) {
+    /// Refreshes [`ClientConnectionQualityScore`] of this [`Connection`].
+    fn refresh_client_conn_quality_score(&self) {
+        use PeerConnectionState as S;
+
         let state = self.0.peer_state.get();
         let quality_score = self.0.quality_score.get();
-
         let score = match (state, quality_score) {
-            (
-                Some(
-                    PeerConnectionState::Disconnected
-                    | PeerConnectionState::Failed,
-                ),
-                _,
-            ) => ClientConnectionQualityScore::Disconnected,
-            (
-                Some(
-                    PeerConnectionState::New | PeerConnectionState::Connecting,
-                ),
-                _,
-            ) => return,
-            (_, Some(quality_score)) => quality_score.into(),
-            _ => return,
```

Comment:
Do not do this, better to cover all cases explicitly, so logic of this those pattern patching will be more clean and more understandable.

Because of this wild-card you're lost `S::Failed` state. If you'd written it without wild card match, then you would see that `Failed` state should be casted to `Disconnected`.